File: Algorithm/Dynamic_Programming/70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<=2){
            return n;
        }
        int a=1;
        int b=2;
        int cur = 0;
        for(int i=2;i<n;i++){
            cur = a+b;
            a=b;
            b=cur;
        }
        return cur;
    }

}

File: Algorithm/Dynamic_Programming/70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp_1 = 1
        dp_2 = 2
        if(n<=2):
            return n
        for i in range(3, n+1):
            dp_n = dp_1 + dp_2
            dp_1 = dp_2
            dp_2 = dp_n
        return dp_n
